holoviews/util/__init__.py

```diff
@@ -105,6 +105,75 @@ def __call__(self, *args, **params):
             return StoreOptions.set_options(obj, options)
 
 
+    @classmethod
+    def expand_options(cls, options, backend=None):
+        """
+        Expands a flat dictionary of options organized by the type
+        of object into the appropriate groups.
+        """
+        current_backend = Store.current_backend
+        loaded_backends = Store.loaded_backends()
+        backend_options = Store.options(backend=backend or current_backend)
+        groups = set(backend_options.groups.keys())
+        expanded = {}
+        for objspec, options in options.items():
+            objtype = objspec.split('.')[0]
+            if objtype not in backend_options:
+                raise ValueError('%s type not found, could not apply options.'
+                                 % objtype)
+            obj_options = backend_options[objtype]
+            expanded[objspec] = {g: {} for g in obj_options.groups}
+            for opt, value in options.items():
+                found = False
+                valid_options = []
+                for g, group_opts in obj_options.groups.items():
+                    if opt in group_opts.allowed_keywords:
+                        expanded[objspec][g][opt] = value
+                        found = True
+                    valid_options += group_opts.allowed_keywords
+                if not found:
+                    if backend is None:
+                        # Check option is invalid for all backends
+                        found = []
+                        for lb in [b for b in loaded_backends if b != 'backend']:
```

Comment:
We spotted this bug together - it should be ``backend`` (a variable) not ``'backend'`` a string. Is there a unit tests that would have caught this bug as the tests did go green....

Reply:
Just went through this, it's not tested because this is just an optimization to ensure it doesn't check the same backend twice.